### ISSUE TITLE
possible difference vs. SciPy for ndimage interpolation with order=0

### ISSUE THREAD

**Issue by grlee77** [cupy/cupy#3661]
The results from `scipy.ndimage` and `cupyx.scipy.ndimage` can occasionally differ at isolated pixel positions when `order=0`. I suspect the cause is due to minor floating point differences that sometimes case the "nearest neighbor" interpolation to use a different image pixel in CuPy vs. SciPy. 

To replicate the issue, you can modify
`test_rotate_float` in `tests/cupyx_tests/scipy_tests/ndimage_tests/test_interpolation.py`, changing the shape from `(10, 10)` to `(32, 32)`.

I get small number of failures in that case with a typical error being:
```
E               AssertionError: 
E               Not equal to tolerance rtol=1e-07, atol=1e-05
E               
E               Mismatched elements: 66 / 30625 (0.216%)
E               Max absolute difference: 9.00128235
E               Max relative difference: 25.86122873
E                x: array([[7.080065, 6.132248, 4.019725, ..., 1.882532, 1.882532, 2.325662],
E                      [7.840422, 2.187558, 7.86325 , ..., 4.017005, 8.584204, 5.419484],
E                      [7.840422, 4.317714, 4.317714, ..., 4.017005, 9.678571, 7.749456],...
E                y: array([[7.080065, 6.132248, 4.019725, ..., 1.882532, 1.882532, 2.325662],
E                      [7.840422, 2.187558, 7.86325 , ..., 4.017005, 8.584204, 5.419484],
E                      [7.840422, 4.317714, 4.317714, ..., 4.017005, 9.678571, 7.749456],...
```

In other words there are 66 pixels out of 30625 where the value is substantially different. Visually the rotated image looks fine, it just appears that at certain locations, the order=0 interpolation landed on a pixel that is off by one from what occurs in SciPy.

I'm not sure there will be any easy way to resolve this as we may just be limited by floating point rounding errors. For now, I would lean toward just documenting the possibile difference in the docstring Notes.
